common/djangoapps/third_party_auth/colaraz.py

```diff
@@ -0,0 +1,60 @@
+from social_core.exceptions import AuthException
+from django.utils.functional import cached_property
+
+from openedx.core.djangoapps.theming.helpers import get_current_request
+from third_party_auth.identityserver3 import IdentityServer3
+
+class ColarazIdentityServer(IdentityServer3):
+    """
+    An extension of the IdentityServer3 for use with Colaraz's IdP service.
+    """
+    name = "colarazIdentityServer"
+    DEFAULT_SCOPE = ["openid", "profile", "IdentityServerApi"]
+    ID_KEY = "email"
+
+    def get_redirect_uri(self, state=None):
+        """
+        Returns redirect uri for oauth redirection
+        """
+        current_req = get_current_request()
+
+        environ = getattr(current_req, "environ", {})
+        schema = environ.get("wsgi.url_scheme", "http")
+
+        site = getattr(current_req, "site", None)
+        domain = getattr(site, "domain", None)
+
+        if not domain:
+            raise AuthException("Colaraz", "Error while authentication")
+
+        return "{}://{}/auth/complete/{}".format(schema, domain, self.name)
+
+    def get_user_details(self, response):
+        """
+        Return details about the user account from the service
+        """
+
+        current_req = get_current_request()
+        site = getattr(current_req, "site", None)
+        domain = getattr(site, "domain", None)
+
+        user_site_domain = response["companyInfo"]["url"]
+
+        if not domain:
+            raise AuthException("Colaraz", "Error while authentication")
+        elif not domain.startswith(user_site_domain.lower()):
+            raise AuthException("Colaraz", "Your account belongs to {}".format(user_site_domain))
+
+        details = {
+            "fullname": "{} {}".format(response["firstName"], response["lastName"]),
+            "email": response["email"],
+            "first_name": response["firstName"],
+            "last_name": response["lastName"],
+            "username": response["email"]
```

Comment:
To remain consistent with edx code base use single quotes where it is possible.

Reply:
Is it really the norm?